### PR TITLE
Add sentry project to track on performance

### LINKED ISSUE #1809: [Quest] Use Sentry to track and monitor performance bottlenecks on quest and FHIR Core
**Name of feature**

Add Sentry performance monitoring to FHIRCore 

**Description of feature**
Basically set up FHIRCore to work with Sentry, and get notified on updates on performance as we continue to add features on FHIRCore.

This would help keep track of performance across the FHIRCore apps, detect high impacts on performance and get breadcrumbs on performance issues

cc @dubdabasoduba

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1809 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/build.gradle
Comment (pld) [inline android/engine/build.gradle:1]: wouldn't we want these changes in the Quest app? That's the overall app

Comment (LZRS) [reply android/engine/build.gradle:1]: We initially had the changes in Quest but ran into an issue whereby we wanted to add custom sentry tracing transactions in Engine to track performance for specific functions involved in sync, and sentry wasn't getting resolved.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/data/register/RegisterPagingSource.kt [hunk 1]
```diff
@@ -72,18 +75,31 @@ class RegisterPagingSource(
...
+          }
+
+        LoadResult.Page(data = data, prevKey = prevKey, nextKey = nextKey)
+      } catch (e: Exception) {
+        transaction.throwable = e
+        transaction.status = SpanStatus.INTERNAL_ERROR
+        throw e
+      } finally {
+        transaction.finish()
+      }
```
[the remark this answers is not in the document]
Comment (LZRS) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/data/register/RegisterPagingSource.kt:102]: Yeah, it was just for testing but will try to use the global handler

File: android/engine/build.gradle
Comment (pld) [reply android/engine/build.gradle:1]: Interesting so Quest can read it when it's in engine but not vice-versa, what about if we use the file in the android dir?

**Comment by pld** [comment]
this looks pretty good @LZRS anything blocking this from merge?

**Comment by LZRS** [comment]
> this looks pretty good @LZRS anything blocking this from merge?

Yeah we were looking to add documentation to guide on how to make use of it in client apps, as part of this PR, and fix codcov patch failing

File: android/quest/src/test/java/org/smartregister/fhircore/quest/QuestApplicationTest.kt [hunk 2]
```diff
@@ -39,6 +44,20 @@ class QuestApplicationTest : RobolectricTest() {
...
+    val spyApp = spyk(application)
+    spyApp.initSentryMonitoring(dsn = sentryDsn)
+  }
+
+  @Test
+  fun testSkipSentryMonitoringWhenDsnBlank() {
+    val sentryDsn = ""
+    val spyApp = spyk(application)
+    spyApp.initSentryMonitoring(dsn = sentryDsn)
+  }
```
Comment (pld) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/QuestApplicationTest.kt:59]: My changes cause these tests to fail for now. If we want to mock, I think we should mock inside SentryAndroid.init, or (not great) we could have that be a default and pass in a mockk for it, but I'd worry that approach would hide problems

**Review by pld — COMMENTED** [review]
can we have all the Timber messages get sent to Sentry?

**Comment by LZRS** [comment]
> can we have all the Timber messages get sent to Sentry?

Yeah, it's possible. We'll set it up as part of this PR

**Comment by LZRS** [comment]
> can we have all the Timber messages get sent to Sentry?

We've confirmed Timber messages are automatically integrated and logged to Sentry, with event level set to _Level.ERROR_

**Review by pld — APPROVED** [review]
great work getting this done!

**Pull request merged**